### PR TITLE
Changed android tests to run against release build

### PR DESCRIPTION
**Pull request by ashwinraghav**

### CODE REVIEW

File: functions/app/build.gradle [hunk 1]
```diff
@@ -15,9 +18,13 @@ android {
     }
 
     buildTypes {
+
+
         release {
             minifyEnabled true
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+            testProguardFiles getDefaultProguardFile('proguard-android.txt'), 'test-proguard-rules.pro'
```
Comment (samtstern) [inline functions/app/build.gradle]: I believe this is missing the final `, 'proguard-rules.pro'` that we discussed

File: functions/app/build.gradle [hunk 2]
```diff
@@ -15,9 +18,13 @@ android {
     }
 
     buildTypes {
+
```
Comment (samtstern) [inline functions/app/build.gradle]: nit: random empty lines? (here and above)

File: functions/app/build.gradle (hunk 1 above)
Comment (ashwinraghav) [reply functions/app/build.gradle]: Later figured that was not required. I was running the wrong variant

**Comment by samtstern** [comment]
Hmmm:

```
* What went wrong:
Could not determine the dependencies of task ':functions:app:check'.
> Task with path 'assembleDebugAndroidTest' not found in project ':functions:app'.
```

**Comment by ashwinraghav** [comment]
Seems like there was a dependency on the assembleDebugAndroidTest that crept into the build.
This is thematic across all other modules too.
It is not clear why this dependency was introduced into the 'check' task in any of the projects since I can't find any actual tests.
What was happening in this case is that setting the testBuildType to release resulted in the "assembleDebugAndroidTest" task no longer being defined. See: https://google.github.io/android-gradle-dsl/current/com.android.build.gradle.AppExtension.html#com.android.build.gradle.AppExtension:testBuildType

**Pull request merged**